cockatrice/src/releasechannel.cpp

```diff
@@ -0,0 +1,307 @@
+}
+#else
+
+bool ReleaseChannel::downloadMatchesCurrentOS(QVariantMap)
+{
+    //If the OS doesn't fit one of the above #defines, then it will never match
+    return false;
+}
+
+#endif
+
+StableReleaseChannel::StableReleaseChannel(int _index, QString _name)
+:ReleaseChannel(_index, _name)
+{
+}
+
+QString StableReleaseChannel::getManualDownloadUrl()
+{
+    return QString(STABLEMANUALDOWNLOAD_URL);
+}
+
+void StableReleaseChannel::checkForUpdates()
+{
+    qDebug() << "Searching for updates on the stable channel: " << STABLERELEASE_URL;
+    response = netMan->get(QNetworkRequest(QString(STABLERELEASE_URL)));
+    connect(response, SIGNAL(finished()), this, SLOT(releaseListFinished()));
+}
+
+void StableReleaseChannel::releaseListFinished()
+{
+    QNetworkReply *reply = static_cast<QNetworkReply *>(sender());
+    bool ok;
+    QString tmp = QString(reply->readAll());
+    reply->deleteLater();
+
+    // Errors are incapsulated in an object, check for them first
+    QVariantMap resultMap = QtJson::Json::parse(tmp, ok).toMap();
+    if (!ok) {
+        qDebug() << "No reply received from the release update server";
+        emit error(tr("No reply received from the release update server."));
+        return;
+    }
+
+    if(!(resultMap.contains("name") && 
+    	resultMap.contains("html_url") && 
+    	resultMap.contains("tag_name") && 
+    	resultMap.contains("published_at"))) {
+        qDebug() << "Invalid received from the release update server";
+        emit error(tr("Invalid reply received from the release update server."));
+        return;
+    }
+
+    if(!lastRelease)
+    	lastRelease = new Release;
+
+	lastRelease->setName(resultMap["name"].toString());
+	lastRelease->setDescriptionUrl(resultMap["html_url"].toString());
+	lastRelease->setPublishDate(resultMap["published_at"].toDate());
+
+    qDebug() << "Got reply from release server, size=" << tmp.size()
+        << "name=" << lastRelease->getName()
+        << "desc=" << lastRelease->getDescriptionUrl()
+        << "date=" << lastRelease->getPublishDate();
+
+    QString url = QString(STABLETAG_URL) + resultMap["tag_name"].toString();
+    qDebug() << "Searching for a corresponding tag on the stable channel: " << url;
+    response = netMan->get(QNetworkRequest(url));
+    connect(response, SIGNAL(finished()), this, SLOT(tagListFinished()));
+}
+
+void StableReleaseChannel::tagListFinished()
+{
+    QNetworkReply *reply = static_cast<QNetworkReply *>(sender());
+    bool ok;
+    QString tmp = QString(reply->readAll());
+    reply->deleteLater();
+
+    QVariantMap resultMap = QtJson::Json::parse(tmp, ok).toMap();
+    if (!ok) {
+        qDebug() << "No reply received from the tag update server";
+        emit error(tr("No reply received from the tag update server."));
+        return;
+    }
+
+    if(!(resultMap.contains("object") && 
+        resultMap["object"].toMap().contains("sha"))) {
+        qDebug() << "Invalid received from the tag update server";
```

Comment:
Is it easy to include the contents of `resultMap` here? Might help. Also probably should be `qWarning` at least